entrypoint.py

```diff
@@ -22,33 +26,35 @@ def _print_cmds(self):
         for i in self.cmds:
             print(i)
 
-    def _get_image(self, img_hash):
-        imgs = self.cli.images()
-        for i in imgs:
-            if img_hash in i['Id']:
-                self.img = i
-                return
-        raise ImageNotFound("Image {} not found\n".format(img_hash))
+    def _get_image(self, image_id):
+        try:
+            return self.client.images.get(image_id)
+        except:
+            raise ImageNotFound("Image {} not found\n".format(image_id))
 
     def _insert_step(self, step):
+        # ignore the end "# buildkit" comment
+        step = re.sub("\s*# buildkit$", "", step)
         if "#(nop)" in step:
-            to_add = step.split("#(nop) ")[1]
+            to_add = step.split("#(nop)")[1].strip()
         else:
-            to_add = ("RUN {}".format(step))
+            # step may contains "/bin/sh -c ", just ignore it
+            to_add = "{}".format(step.replace("/bin/sh -c ", ""))
```

Comment:
```suggestion
            to_add = "RUN {}".format(step.replace("/bin/sh -c ", ""))
```

Reply:
If add `RUN ` here, some docker commands result redundant `RUN ` output.

<img width="958" alt="image" src="https://github.com/lukapeschke/dockerfile-from-image/assets/2276718/ab9bd09d-6334-43e4-abee-30f61f5aafb0">

Reply:
Hmm that's really weird :thinking: , I can't reproduce it on my side... I've noted myself to add some proper versioning + CI when I have the time: https://github.com/lukapeschke/dockerfile-from-image/issues/4